### DISCUSSION TITLE
Some ninja spinners are automatically completed in an old beatmap

### DISCUSSION THREAD

**Discussion by Gabixel** [ppy/osu#20182] [Q&A]
(If I am not mistaken, the spinners in this map are the so-called "[ninja spinners](https://osu.ppy.sh/wiki/en/Beatmapping/Mapping_techniques/Unrankable#ninja-spinners)". I apologize if I used the wrong term.)

For some reason, [in this 14-year-old map](https://osu.ppy.sh/beatmapsets/2201#osu/19043), these specific ninja spinners toward the end of the map are automatically completed:

- Seen by the player (I used `NoFail` to show you that I did not press any keys throughout the video):
  
  https://user-images.githubusercontent.com/43073074/188977433-cf1dbb02-637c-4a05-8025-093229a7e358.mp4

- Seen by `AutoPlay` (seems to ignore them totally):
  
  https://user-images.githubusercontent.com/43073074/188977493-ba96860b-760a-4409-90b6-46fb3385df87.mp4

**Comment by smoogipoo** [comment]
Yes, spinners that are too short are automatically completed.

How does this play in stable?

**Reply by Gabixel** [reply]
In stable, all spinners in this map give a 50 ("Meh") if untouched, and the spinners in question in the videos above give a 100 ("OK") instead.
AutoPlay seems to *not* ignore them and thus spins them correctly.

**Comment by cihe13375** [comment]
Another example of this issue: https://osu.ppy.sh/beatmapsets/1615177#osu/3708312 (ranked in 2022).

There are 79 spinners in the map.
On stable, 39 of them give 50s and 39 give 100s if the player does nothing.
On lazer, 39 of them give 300s and others give misses. Btw, autoplay seems to ignore some of the "300" spinners completely (the cursor doesn't move at all).

I think we shall make stable and lazer match each other so players can compete fairly on this kind of maps after the leaderboard is merged.

Also it should probably be considered for spinner difficulty calculation (https://github.com/ppy/osu/pull/21621), e.g. a spinner should not add difficulty at all if it gives 300 automatically.

**Reply by bdach** [reply]
> I think we shall make stable and lazer match each other so players can compete fairly on this kind of maps after the leaderboard is merged.

As a team we're pretty fine with lazer being easier than stable in some respects. To that end I'm not particularly inclined to dig into whatever fresh hell stable was doing with spinners for a sake of few maps, especially so that lazer mechanics are more forgiving.

**Reply by cihe13375** [reply]
I'm fine with either, but please note that in this case lazer is not always easier than stable (the 50s in stable becomes misses in lazer)
